---
int32 row
int32 col
float64[] waypoints